### PR TITLE
Fixes crash when there's a missing texture argument

### LINKED ISSUE #6484: Missing WGSL texture argument crashes Naga
**Description**
This shader crashes Naga:
```wgsl
@group(0) @binding(0) var my_sampler: sampler;

fn foo(tex: texture_2d<f32>) -> vec4<f32> {
    return textureSampleLevel(tex, my_sampler, vec2f(0, 0), 0.0);
}

fn main() {
    foo();
}
```

**Repro steps**
1. Create `test.wgsl` containing the shader above 
2. Execute `naga test.wgsl`

**Expected vs observed behavior**

An error message is expected explaining the missing texture argument in the `foo()` function call.
Instead, the following panic message is seen:
```
naga crash_texture.wgsl
thread 'main' panicked at C:\Users\korte\.cargo\registry\src\index.crates.io-6f17d22bba15001f\naga-22.1.0\src\valid\analyzer.rs:424:34:
index out of bounds: the len is 0 but the index is 0
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Platform**
Windows 11
Naga 22.0.0 (according to `naga --version`)

### PR DESCRIPTION
**Pull request by aedm**
**Connections**
Fixes https://github.com/gfx-rs/wgpu/issues/6484

**Description**
There was an OOB when processing texture/sampler combinations across function calls.

**Testing**
Added a unit test checking for a proper validation error message.

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `taplo format`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown`
  - [x] `--target wasm32-unknown-emscripten`
- [x] Run `cargo xtask test` to run tests.
- [x] Add change to `CHANGELOG.md`. See simple instructions inside file.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -221,6 +221,7 @@ By @MarijnS95 in [#6006](https://github.com/gfx-rs/wgpu/pull/6006).
 - Matrices can now be indexed by value (#4337), and indexing arrays by value no longer causes excessive spilling (#6358). By @jimblandy in [#6390](https://github.com/gfx-rs/wgpu/pull/6390).
 - Add support for `textureQueryLevels` to the GLSL parser. By @magcius in [#6325](https://github.com/gfx-rs/wgpu/pull/6415).
 - Fix unescaped identifiers in the Metal backend shader I/O structures causing shader miscompilation. By @ErichDonGubler in [#6438](https://github.com/gfx-rs/wgpu/pull/6438).
+- Fix crash when a texture argument is missing. By @aedm in [#6486](https://github.com/gfx-rs/wgpu/pull/6486)
```
Comment (Wumpf) [inline CHANGELOG.md]: this should go under `## Unreleased`, not to an release that's already out

Comment (aedm) [reply CHANGELOG.md]: Fixed.

**Comment by aedm** [comment]
@Wumpf Request fixed, please take a look.

**Review by teoxoy — APPROVED** [review]
Thanks for the fix!

**Comment by teoxoy** [comment]
@aedm clippy is unhappy

**Comment by aedm** [comment]
@teoxoy Sorry, Clippy is happy now.

**Pull request merged**